src/meta/processors/job/DataBalanceJobExecutor.cpp

```diff
@@ -0,0 +1,197 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+#include "meta/processors/job/DataBalanceJobExecutor.h"
+
+#include <folly/executors/CPUThreadPoolExecutor.h>
+
+#include "common/utils/MetaKeyUtils.h"
+#include "kvstore/NebulaStore.h"
+#include "meta/processors/job/JobUtils.h"
+
+namespace nebula {
+namespace meta {
+
+folly::Future<Status> DataBalanceJobExecutor::executeInternal() {
+  if (plan_ == nullptr) {
+    Status status = buildBalancePlan();
+    if (status != Status::OK()) {
+      return status;
+    }
+  }
+  plan_->setFinishCallBack([this](meta::cpp2::JobStatus status) {
+    if (LastUpdateTimeMan::update(kvstore_, time::WallClock::fastNowInMilliSec()) !=
+        nebula::cpp2::ErrorCode::SUCCEEDED) {
+      LOG(ERROR) << "Balance plan " << plan_->id() << " update meta failed";
+    }
+    executorOnFinished_(status);
+  });
+  plan_->invoke();
+  return Status::OK();
+}
+
+Status DataBalanceJobExecutor::buildBalancePlan() {
+  std::map<std::string, std::vector<Host*>> lostZoneHost;
+  std::map<std::string, std::vector<Host*>> activeSortedHost;
+  for (auto& zoneMapEntry : spaceInfo_.zones_) {
```

Comment:
could use `auto& [zname, zinfo]` or sth like that, instead of entry.second? the previous one is more readable. Because we could not figure it out what does the entry.second means.

Reply:
done